service/application/src/main/java/com/solace/maas/ep/event/management/agent/subscriber/SolacePersistentMessageHandler.java

```diff
@@ -68,21 +91,46 @@ public void onMessage(InboundMessage inboundMessage) {
             log.trace("onMessage: {}\n{}", messageClass, messageAsString);
             message = toMessage(messageAsString, messageClass);
             processor.processMessage(processor.castToMessageClass(message));
-
+            if (messageHandlerObserver != null) {
+                messageHandlerObserver.onMessageProcessingCompleted(inboundMessage);
+            }
         } catch (Exception e) {
-            if (processor != null && message != null) {
-                log.error("Error while processing inbound message from queue for mopMessageSubclass: {}", mopMessageSubclass);
-                try {
-                    processor.onFailure(e, processor.castToMessageClass(message));
-                } catch (Exception e1) {
-                    log.error("error while handling message processing failure for mopMessageSubclass: {}", mopMessageSubclass, e);
-                }
-
-            } else {
-                log.error("Unsupported message and/or processor encountered. Skipping processing", e);
+            handleProcessingError(mopMessageSubclass, processor, message, e);
+            if (messageHandlerObserver != null) {
+                messageHandlerObserver.onMessageProcessingFailed(inboundMessage);
             }
-
         } finally {
+            acknowledgeMessage(inboundMessage);
+            if (messageHandlerObserver != null) {
+                messageHandlerObserver.onMessageProcessingAcknowledged(inboundMessage);
+            }
```

Comment:
This not null check in multiple places seems a bit odd. Can we create a private method such as 
```
private void handleMessageHandlerObserver(InboundMessage inboundMessage, InboundMessageLifecycle     InboundMessageLifecycle){
  if(messageHandlerObserver == null) {
    return;
  }
if(InboundMessageLifecycle == InboundMessageLifecycle.MessageReceived){
   messageHandlerObserver.onMessageReceived(inboundMessage);
}
// and so on
}
```
You need to create an Enum class for InboundMessageLifecycle